Tools/SetupUnityPackage/UnityUtils/FileHelper.cs

```diff
@@ -33,16 +33,22 @@ public static string GetInternalStorage()
 
             // On Android persistentDataPath returns the external folder, where the app may not have permissions to write.
             // we use reflection to call File.getAbsolutePath(currentActivity.getFilesDir())
-            using (var unityPlayer = new AndroidJavaClass("com.unity3d.player.UnityPlayer"))
-            using (var currentActivity = unityPlayer.GetStatic<AndroidJavaObject>("currentActivity"))
-            {
-                var getFilesDir = AndroidJNIHelper.GetMethodID(AndroidJNI.FindClass("android/content/ContextWrapper"), "getFilesDir", "()Ljava/io/File;");
-                var internalFilesDir = AndroidJNI.CallObjectMethod(currentActivity.GetRawObject(), getFilesDir, Array.Empty<jvalue>());
-                var getAbsolutePath = AndroidJNIHelper.GetMethodID(AndroidJNI.FindClass("java/io/File"), "getAbsolutePath", "()Ljava/lang/String;");
+            var path = string.Empty;
 
-                var path = AndroidJNI.CallStringMethod(internalFilesDir, getAbsolutePath, Array.Empty<jvalue>());
-                return path ?? $"/data/data/{Application.identifier}/files";
+            try
```

Comment:
We don't need a try-catch here - the consumer will handle the exceptions ([it just suppresses them](https://github.com/realm/realm-dotnet/blob/64bfe35ff4226a3e0c18b5db361ab28a8e36a775/Realm/Realm/InteropConfig.cs#L100-L117), but we may eventually want to log them somehow).

Reply:
Should I look into this already? If so, what did you have in mind? A logfile or something else?

Reply:
No, that's outside of the scope of this change. It was more of a recommendation to always let the final caller handle exceptions.